pkg/client/results/processing.go

```diff
@@ -0,0 +1,181 @@
+/*
+Copyright the Sonobuoy contributors 2019
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package results
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/heptio/sonobuoy/pkg/plugin"
+	"github.com/heptio/sonobuoy/pkg/plugin/driver/daemonset"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// StatusFailed is the key we base junit pass/failure off of and save into
+	// our canonical results format.
+	StatusFailed = "failed"
+
+	// StatusPassed is the key we base junit pass/failure off of and save into
+	// our canonical results format.
+	StatusPassed = "passed"
+
+	// StatusSkipped is the key we base junit pass/failure off of and save into
+	// our canonical results format.
+	StatusSkipped = "skipped"
+
+	// StatusUnknown is the key we fallback to in our canonical results format
+	// if another can not be determined.
+	StatusUnknown = "unknown"
+
+	// PostProcessedResultsFile is the name of the file we create when doing
+	// postprocessing on the plugin results.
+	PostProcessedResultsFile = "sonobuoy_results.yaml"
+)
+
+// postProcessor is a function which takes two strings: the plugin directory and the
+// filepath in question, and parse it to create an Item.
+type postProcessor func(string, string) (Item, error)
+
+// fileSelector is a type of a function which, given a filename and the FileInfo will
+// determine whether or not that file should be postprocessed. Allows matching a specific
+// file only or all files with a given suffix (for instance).
+type fileSelector func(string, os.FileInfo) bool
+
+// aggregateStatus defines the aggregation rules for status. Failures bubble
+// up and otherwise the status is assumed to pass.
+func aggregateStatus(items ...Item) string {
+	for i, v := range items {
+		// Branches should just aggregate their leaves and return the result.
+		if len(v.Items) > 0 {
+			items[i].Status = aggregateStatus(v.Items...)
+			return items[i].Status
+		}
+
+		// Any failures immediately fail the parent.
+		if v.Status == StatusFailed {
+			return StatusFailed
+		}
+	}
+
+	// Only pass if no failures found.
+	return StatusPassed
+}
+
+func PostProcessPlugin(p plugin.Interface, dir string) (Item, error) {
+	i := Item{}
+	var err error
+
+	switch p.GetResultFormat() {
+	case "junit", "e2e":
```

Comment:
Perhaps these could be const values? Also, will there be any list of supported result types?

Reply:
Added constants; didnt exactly add a "list" explicitly, but the supported types are just the ones that we have constants for. Our documentation will reference those.